pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -783,6 +796,19 @@ func (c *Cluster) getActivateOSDInitContainer(configDir, namespace, osdID string
 	return volume, container
 }
 
+func (c *Cluster) getnewkeyring(osdId string, osdProps osdProperties) v1.Container {
+	return v1.Container{
+		Name:  getkeyring,
+		Image: c.spec.CephVersion.Image,
+		Command: []string{
+			"/bin/bash",
+			"-c",
+			fmt.Sprintf(newkeyring, fmt.Sprintf("%s", osdId)),
+		},
+		Resources: osdProps.resources,
```

Comment:
The container might be missing the mounts required to run this command. `/etc/ceph` and/or `/etc/rook`. These are used in toolbox container.